census/census.go

```diff
@@ -82,6 +105,18 @@ func (m *Manager) Init(storageDir, rootAuthPubKey string) error {
 	m.AuthWindow = 10
 	m.compressor = newCompressor()
 
+	dbDir := filepath.Join(m.StorageDir, fmt.Sprintf("v%v", CensusVersion))
+	db, err := badgerdb.New(badgerdb.Options{Path: dbDir})
+	if err != nil {
+		return err
+	}
+	m.db = &BadgerDBBatched{db}
+	defer func() {
+		if err != nil {
```

Comment:
@mvdan My intention here is to handle the case where the function Init returns err != nil.  I believe this will work, because in the function scope, the variable "err" refers to the return value, and on any return, whatever is returned is assigned to err, and afterwards the defer is called.  Here are some test that make me think this works: https://play.golang.org/p/2Hj7inmOWV5